website/content/docs/concepts/cpu.mdx

```diff
@@ -0,0 +1,324 @@
+CPU.
+
+```
+bandwidth = (p_cores * p_frequency) + (e_cores * e_frequency)
+```
+
+The total number of cores is computed by summing the number of P-Cores and the
+number of E-Cores.
+
+```
+cores = p_cores + e_cores
+```
+
+Nomad does not distinguish between logical and physical CPU cores. One of the
+defining differences between the P-Core and E-Core types is that the E-Cores
+do not support hyperthreading, whereas P-Cores do. As such a single physical
+P-Core is presented as 2 logical cores, and a single E-Core is presented as 1
+logical core.
+
+The example below is from a Nomad node with an Intel i9-13900 CPU. It is made
+up of mixed core types, with a P-Core base frequency of 2 GHz and an E-Core
+base frequency of 1.5 GHz.
+
+These characteristics are reflected in the `cpu.frequency.performance` and
+`cpu.frequency.efficiency` node attributes respectively. 
+
+```text
+cpu.arch                        = amd64
+cpu.frequency.efficiency        = 1500
+cpu.frequency.performance       = 2000
+cpu.modelname                   = 13th Gen Intel(R) Core(TM) i9-13900
+cpu.numcores                    = 32
+cpu.numcores.efficiency         = 16
+cpu.numcores.performance        = 16
+cpu.reservablecores             = 32
+cpu.totalcompute                = 56000
+cpu.usablecompute               = 56000
+```
+
+## Reserving CPU Resources
+
+In the fingerprinted node attributes, `cpu.totalcompute` indicates the total
+amount of CPU bandwidth the processor is capable of delivering. In some cases it
+may be beneficial to reserve some amount of a node's CPU resources for use by
+the operating system and other non-Nomad processes. This can be done in client
+configuration.
+
+The amount of reserved CPU can be specified in bandwidth via `cpu`.
+
+```hcl
+client {
+  reserved {
+    cpu = 3000 # mhz
+  }
+}
+```
+
+Or as a specific set of `cores` on which to disallow the scheduling of Nomad
+tasks. This capability is available on Linux systems only.
+
+```hcl
+client {
+  reserved {
+    cores = "0-3"
+  }
+}
+```
+
+When the CPU is constrained by one of the above configurations, the node
+attribute `cpu.usablecompute` indicates the total amount of CPU bandwdith
+available for scheduling of Nomad tasks.
+
+## Allocating CPU Resources
+
+When scheduling jobs, a Task must specify how much CPU resource should be
+allocated on its behalf. This can be done in terms of bandwidth in MHz
+with the `cpu` attribute.
+
+```hcl
+task {
+  resources {
+    cpu = 2000 # mhz
+  }
+}
+```
+
+Note that the isolation mechansim around CPU resources is dependent on each
+task driver and its configuration. The standard behavior is that Nomad ensures
+a task has access to _at least_ as much of its allocated CPU bandwidth. In which
+case if a node as idle CPU capacity, a task may use additional CPU resources.
+Some task drivers enable limiting a task to use only the amount of bandwidth
+allocated to the task, described in the CPU Hard Limits section below.
```

Comment:
It might be nice if we specifically referred to the typical control mechanism here, which is mapping Mhz to cpu shares 1:1.